ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -313,26 +313,74 @@ private void executeCommands(
       final Clock clock,
       final String previous
   ) {
-    final Map<String, Object> properties = new HashMap<>();
     final List<String> commands = CommandParser.splitSql(migrationFileContent);
 
+    executeCommands(
```

Comment:
Add a comment explaining what this does: it doesn't actually execute anything, just steps through the commands to ensure that they parse (before the next call which actually executes statements). It'd be good for the comment to explain why it's necessary to "execute" DEFINE and UNDEFINE statements as part of this parsing validation as well.